graphql-server/src/schemas/neo4j.ts

```diff
@@ -51,64 +55,36 @@ function buildResolvers(ogm: OGM, apolloClient: typeof ApolloClient) {
-            where.hasMetadataSampleMetadataConnection_SOME.node.primaryId,
-          { json: true }
+        const primaryId =
+          where.hasMetadataSampleMetadataConnection_SOME.node.primaryId;
+
+        const sampleManifest = await request(
+          props.smile_sample_endpoint + primaryId,
+          {
+            json: true,
+          }
         );
 
-        const smdataupdates = update.hasMetadataSampleMetadata[0].update.node;
-        Object.keys(smdataupdates).forEach((key: string) => {
-          data[key] = smdataupdates[key];
-        });
-
-        // remove 'status' from sample metadata to ensure validator and label
-        // generator use latest status data added during validation process
-        delete data["status"];
-
-        // add isCmoSample to sample's 'additionalProperties' if not already present
-        // this is to ensure that cmo samples get sent to the label generator after validation
-        // since some of the older SMILE samples do not have this additionalProperty set
-        if (data["additionalProperties"]["isCmoSample"] == null) {
-          const requestId = data["additionalProperties"]["igoRequestId"];
-          let req = ogm.model("Request");
-          const rd = await req.find({
-            where: { igoRequestId: requestId },
-          });
-          data["additionalProperties"]["isCmoSample"] =
-            rd[0]["isCmoRequest"].toString();
-        }
+        const sampleKeyForUpdate = Object.keys(update)[0];
 
-        // fire and forget
-        publishNatsMessage(
-          props.pub_validate_sample_update,
-          JSON.stringify(data)
-        );
+        const changesByPrimaryId = update[sampleKeyForUpdate][0].update.node;
 
-        let sample = ogm.model("Sample");
-
-        await sample.update({
-          where: { smileSampleId: data.smileSampleId },
-          update: { revisable: false },
-        });
-
-        const updatedSamples = await apolloClient.query({
-          query: SamplesDocument,
-          variables: {
-            where: {
-              smileSampleId: data.smileSampleId,
-            },
-            hasMetadataSampleMetadataOptions2: {
-              sort: [{ importDate: SortDirection.Desc }],
-              limit: 1,
-            },
-          },
-        });
-
-        Object.keys(smdataupdates).forEach((key: string) => {
-          updatedSamples.data.samples[0].hasMetadataSampleMetadata[0][key] =
-            smdataupdates[key];
-        });
+        let updatedSamples: any;
+        if ("hasMetadataSampleMetadata" in update) {
+          updatedSamples = await updateSampleMetadata(
+            sampleManifest,
+            changesByPrimaryId,
+            ogm,
+            apolloClient
+          );
+        } else {
+          updatedSamples = await updateSampleBilling(
+            sampleManifest,
+            primaryId,
+            changesByPrimaryId,
+            apolloClient
+          );
+        }
 
```

Comment:
Most of the code changes here were me moving the existing code into a named function updateSampleMetadata(), so that we can enable this resolver to handle updating billing data as well (see function updateSampleBilling()).